jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -859,8 +859,15 @@ protected void doStart() throws Exception
             throw new IllegalStateException("Null contextPath");
 
         if (getBaseResource() != null && getBaseResource().isAlias())
+        {
+            // We may have symlink to baseResource, try to resolve symlink if possible.
+            File file = getBaseResource().getFile();
+            if (file != null)
+                _baseResource = Resource.newResource(file.toPath().toRealPath());
+
```

Comment:
I'm pretty sure the WebAppProvider in the deployer does something similar.   Perhaps wrap up in a utlility static method on Resource

Reply:
I added a static method on `Resource` called `resolveAlias`.

Reply:
yeah but it needs javadoc!